### DISCUSSION TITLE
Zombie children when using React 16?

### DISCUSSION THREAD

**Discussion by antoniom** [pmndrs/zustand#1039] [Q&A] [answered]
Forgive me if the "zombie children" term is not correct, but from what I understand this is what I figured out while researching about the issue.
It seems that zustand's selectors are executing in incorrect order after a specific scenario.

Imagine the following application:

```
App
 |-- View1
      |-- Viewer
      |-- Editor
 |-- View2
```

- There is a ternary in `App` which renders either `View1` or `View2`
- There is a ternary in `View1` which looks up a zustand store and decides which component (`Viewer` or `Editor`) to render
- The `Viewer` contains a button which updates zustand store so that we can transit to `Editor`
- The `Editor` contains a controlled input where its value is stored in zustand. The `Editor` component has a "submit" button which erases that value, thus making the transition to `Viewer`
 
 Code sandbox link: https://codesandbox.io/s/sharp-pike-zdomm2

Perform the following steps:
1. Click on button "Show view 1":
2. Click on button "EDIT"
3. Click on button "Show view 2":
4. Click on button "Show view 1":
5. Type anything on the field and hit "SUBMIT" button

Actual behaviour: Receiving type error.
Expected behaviour: switch back to View component without any error.

Notes: 
1) Issue is not reproducible when omitting steps 3 and 4
2) Issue is not reproducible when using React 18

I suspect that when returning back to View1 useStore's selectors are not placed in the correct order, but how can we correct this ?

**Comment by dai-shi** [comment, marked as answer]
If it's about zombie children, please try "Calling actions outside a React event handler in pre React 18" in [readme](https://github.com/pmndrs/zustand#best-practices).

**Reply by antoniom** [reply]
@dai-shi thanks! Although I read it in docs I didn't realize that this was my case
